### PR TITLE
Aftershock errors to be fixed

### PR DESCRIPTION
**Pull request by Maleclypse**
#### Summary
SUMMARY: Bugfixes "Remove aftershock from mod test blacklist"

#### Purpose of change
Removing mods from the testing blacklist.

#### Describe the solution
KorGgenT sent me a file of failed tests and I am working on resolving them.  

#### Describe alternatives you've considered
None

#### Testing
[Test results AFS.txt](https://github.com/CleverRaven/Cataclysm-DDA/files/3999544/Test.results.AFS.txt)

I've marked clear on any that I think I've resolved so far.

### CODE REVIEW

**Comment by Maleclypse** [comment]
@I-am-Erk can you help with where I should be putting `mon_faction` to align an npc with a group of monsters?  `09:36:02.974 ERROR : d:\projects\cataclysm-dda-fork\src\json.cpp:112: Failed to visit member 'mon_faction' in JsonObject at line 14:6:
{
    "type": "npc",
    "id": "afs_gardener_cyborg",
    "//": "PrepNet Phyle Gardener.",
    "name_suffix": "Gardener",
    "class": "afs_prepnet_gardener",
    "attitude": 1,
    "mission": 0,
    "chat": "TALK_PrepNet_gardener_1",
    "faction": "Prepnet_Phyle",
    "mon_faction": "PrepNet"
  }`

**Comment by Maleclypse** [comment]
@KorGgenT  I'm not understanding this error `09:36:02.983 ERROR : d:\projects\cataclysm-dda-fork\src\json.cpp:112: Failed to visit member 'entries' in JsonObject at line 40:1995:
 {
      "both": [ "laptop", "software_useless", "linuxtshirt", "pants_cargo", "socks", "slingpack", "sneakers", "smart_phone" ],
      "entries": [ { "item": "light_plus_battery_cell", "ammo-item": "battery", "charges": 150, "container-item": "eink_tablet_pc" } ],
      "male": [ "briefs" ],
      "female": [ "bra", "panties" ]
    }` 
since the profession 
`  {
    "type": "profession",
    "ident": "afs_linux_patron",
    "name": "Linux Patron",
    "description": "You've grown up using Linux, have hacked together your own operating system out of Gentoo and a bunch of packages, and campaign for open source like it's a basic human right.  None of these things matter in the apocalypse, but your knowledge of the inner workings of 87% of the world's servers might.",
    "points": 2,
    "skills": [ { "level": 5, "name": "computer" } ],
    "items": {
      "both": [ "laptop", "software_useless", "linuxtshirt", "pants_cargo", "socks", "slingpack", "sneakers", "smart_phone" ],
      "entries": [ { "item": "light_plus_battery_cell", "ammo-item": "battery", "charges": 150, "container-item": "eink_tablet_pc" } ],
      "male": [ "briefs" ],
      "female": [ "bra", "panties" ]
    }`
uses `entries` in the same way that other professions not popping errors seem to be using it.

**Comment by Maleclypse** [comment]
@tenmillimaster using `clip_size` for lasers appears to cause a fatal error. I'd like to keep it a laser for now but I suppose I can create some kind of ammo for it. Or I could look at wands from Magiclysm and see if I can make a spell that it casts with charges? Would that make aftershock require magiclysm?

**Comment by KorGgenT** [comment]
that would not make aftershock require magiclysm

**Comment by I-am-Erk** [comment]
> @I-am-Erk can you help with where I should be putting `mon_faction` to align an npc with a group of monsters? `09:36:02.974 ERROR : d:\projects\cataclysm-dda-fork\src\json.cpp:112: Failed to visit member 'mon_faction' in JsonObject at line 14:6: { "type": "npc", "id": "afs_gardener_cyborg", "//": "PrepNet Phyle Gardener.", "name_suffix": "Gardener", "class": "afs_prepnet_gardener", "attitude": 1, "mission": 0, "chat": "TALK_PrepNet_gardener_1", "faction": "Prepnet_Phyle", "mon_faction": "PrepNet" }`
> 
> Whenever you get to it, I totally get it's the holidays :)

I'm not completely sure, I would start by checking out [monster factions](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/data/json/monster_factions.json) and just seeing what happens if you give an npc a monster faction. Whatever is done for Aspis, the bee NPC, is probably what will work.

**Comment by Maleclypse** [comment]
> I'm not completely sure, I would start by checking out [monster factions](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/data/json/monster_factions.json) and just seeing what happens if you give an npc a monster faction. Whatever is done for Aspis, the bee NPC, is probably what will work.

Apis doesn't have a `mon_faction`.  I've searched `mon_faction" and `apis` to see if I'm misreading it. 
The only place that uses mon_faction in vanilla is `your_followers`. I've mirrored that in my update.

File: data/mods/Aftershock/player/afs_professions.json [hunk 1]
```diff
@@ -33,9 +33,9 @@
     "skills": [ { "level": 5, "name": "computer" } ],
     "items": {
       "both": {
-      "itmes": [ "laptop", "software_useless", "linuxtshirt", "pants_cargo", "socks", "slingpack", "sneakers", "smart_phone" ],
-      "entries": [ { "item": "light_plus_battery_cell", "ammo-item": "battery", "charges": 150, "container-item": "eink_tablet_pc" } ]
-    },
+        "itmes": [ "laptop", "software_useless", "linuxtshirt", "pants_cargo", "socks", "slingpack", "sneakers", "smart_phone" ],
```
Comment (Zireael07) [inline data/mods/Aftershock/player/afs_professions.json]: I imagine it should be "items" not "itmes"?

Comment (Maleclypse) [reply data/mods/Aftershock/player/afs_professions.json]: Thanks for catching that!

**Comment by Maleclypse** [comment]
@anothersimulacrum 
I'm wanting it to deconstruct from a corpse as a non reloadable gun with charges.  `[   {     "result": "broken_afs_mon_migoturret",     "type": "uncraft",     "skill_used": "survival",     "difficulty": 5,     "time": "1 h",     "using": [ [ "soldering_standard", 10 ] ],       "components": [       [ [ "living_brain_jar", 1 ] ],       [ [ "brain", 4 ] ],       [ [ "mutant_meat", 12 ] ],       [ [ "vibrating_blaster", 1 ], "charges": { 70, 80 } ],       [ [ "humming_heart", 1 ] ],       [ [ "sensory_cluster", 1 ] ],       [ [ "bioweapon_chassis", 1 ] ]     ]   }`

**Comment by Maleclypse** [comment]
@ZhilkinSerg I saw where you made bugfixes earlier today and I needed to resolve conflicts with those.  I tried to make sure I did them correctly but I would appreciate it if you could review this when you have a moment.   Thanks!

File: data/mods/Aftershock/items/afs_tools.json [hunk 2]
```diff
@@ -19,9 +19,21 @@
...
+      "MP3"
     ],
-    "flags": [ "WATCH", "ALARMCLOCK" ]
+    "flags": [ "WATCH", "ALARMCLOCK", "NO_UNLOAD", "NO_RELOAD" ]
+  },
+  {
+    "id": "afs_atomic_smartphone_music",
+    "copy-from": "afs_atomic_smartphone",
+    "type": "TOOL",
+    "name": "aomic smartphone - music",
```
Comment (ZhilkinSerg) [inline data/mods/Aftershock/items/afs_tools.json]:
```suggestion
    "name": "atomic smartphone - music",
```

File: data/mods/Aftershock/items/afs_bioparts.json [hunk 3]
```diff
@@ -90,23 +90,25 @@
     "type": "GUN",
     "name": "vibrating bioblaster",
     "name_plural": "bioblasters",
```
Comment (Qrox) [inline data/mods/Aftershock/items/afs_bioparts.json]: Should the plural form also have "vibrating"?

Comment (Maleclypse) [reply data/mods/Aftershock/items/afs_bioparts.json]: You are correct and I needed to change some descriptions to better match other changes.

**Pull request merged**